pinot-core/src/main/java/com/linkedin/pinot/core/util/AvroUtils.java

```diff
@@ -16,138 +16,120 @@
-   */
-  private static Map<String, FieldSpec.FieldType> getDefaultFieldTypes(org.apache.avro.Schema avroSchema) {
-    Map<String, FieldSpec.FieldType> fieldTypes = new HashMap<>();
-
-    for (final Field field : avroSchema.getFields()) {
-      FieldSpec.FieldType fieldType;
-
-      if (field.name().contains(COUNT) || field.name().contains(METRIC)) {
-        fieldType = FieldSpec.FieldType.METRIC;
-      } else if (field.name().contains(DAY) || field.name().equalsIgnoreCase(DAYS_SINCE_EPOCH)) {
-        fieldType = FieldSpec.FieldType.TIME;
-      } else {
-        fieldType = FieldSpec.FieldType.DIMENSION;
-      }
-
-      fieldTypes.put(field.name(), fieldType);
-    }
-
-    return fieldTypes;
-  }
-
   /**
-   * Given an avro schema object along with column field types and time unit, return the equivalent
-   * pinot schema object.
+   * Given an Avro schema, map from column to field type and time unit, return the equivalent Pinot schema.
    *
-   * @param avroSchema Avro schema for which to get the Pinot schema.
-   * @param fieldTypes Map containing fieldTypes for each column.
-   * @param timeUnit Time unit to be used for the time column.
-   * @return Return the equivalent pinot schema for the given avro schema.
+   * @param avroSchema Avro schema
+   * @param fieldTypeMap Map from column to field type
+   * @param timeUnit Time unit
+   * @return Pinot schema
+   * @throws IOException
    */
-  private static Schema getPinotSchemaFromAvroSchema(org.apache.avro.Schema avroSchema,
-      Map<String, FieldSpec.FieldType> fieldTypes, TimeUnit timeUnit) {
+  public static Schema getPinotSchemaFromAvroSchema(@Nonnull org.apache.avro.Schema avroSchema,
+      @Nullable Map<String, FieldSpec.FieldType> fieldTypeMap, @Nullable TimeUnit timeUnit) {
     Schema pinotSchema = new Schema();
 
     for (Field field : avroSchema.getFields()) {
       String fieldName = field.name();
       FieldSpec.DataType dataType = extractFieldDataType(field);
       boolean isSingleValueField = isSingleValueField(field);
-
-      FieldSpec.FieldType fieldType = fieldTypes.get(fieldName);
-      switch (fieldType) {
-        case DIMENSION:
-          pinotSchema.addField(new DimensionFieldSpec(fieldName, dataType, isSingleValueField));
-          break;
-        case METRIC:
-          Preconditions.checkState(isSingleValueField, "Unsupported multi-value for metric field.");
-          pinotSchema.addField(new MetricFieldSpec(fieldName, dataType));
-          break;
-        case TIME:
-          Preconditions.checkState(isSingleValueField, "Unsupported multi-value for time field.");
-          pinotSchema.addField(new TimeFieldSpec(field.name(), dataType, timeUnit));
-          break;
-        default:
-          throw new UnsupportedOperationException("Unsupported field type: " + fieldType + " for field: " + fieldName);
+      if (fieldTypeMap == null) {
+        pinotSchema.addField(new DimensionFieldSpec(fieldName, dataType, isSingleValueField));
```

Comment:
Can we log a message here, so user knows we are treating these columns as dimensions?

Reply:
This happens only when we call it with null fieldTypeMap (API: getPinotSchemaFromAvroDataFile(@Nonnull File avroDataFile)), and the java doc describes the behavior.